druid/examples/async_event.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2020 The Druid Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An example of sending commands from another thread.
+//! This is useful when you want to have some kind of
+//! generated content (like here), or some task that just
+//! takes a long time but don't want to block the main thread
+//! (waiting on an http request, some cpu intensive work etc.)
+
+use instant::Instant;
+use std::thread;
+use std::time::Duration;
+
+use druid::widget::prelude::*;
+use druid::{AppLauncher, Color, Selector, Target, WidgetExt, WindowDesc};
+
+// If you want to submit commands to an event sink you have to give it some kind
+// of ID. The selector is that, it also assures the accompanying data-type is correct.
+// look at the docs for `Selector` for more detail.
+const SET_COLOR: Selector<Color> = Selector::new("event-example.set-color");
+
+pub fn main() {
+    let window = WindowDesc::new(make_ui).title("External Event Demo");
+
+    let launcher = AppLauncher::with_window(window);
+
+    // If we want to create commands from another thread `launcher.get_external_handle()`
+    // should be used. For sending commands from within widgets you can always call
+    // `ctx.submit_command`
+    let event_sink = launcher.get_external_handle();
+    // We create a new thread and generate colours in it.
+    // This happens on a second thread so that we can run the UI in the
+    // main thread. Generating some colours nicely follows the pattern for what
+    // should be done like this: generating something over time
+    // (like this or reacting to external events), or something that takes a
+    // long time and shouldn't block main UI updates.
+    thread::spawn(move || generate_colours(event_sink));
+
+    launcher
+        .use_simple_logger()
+        .launch(Color::BLACK)
+        .expect("launch failed");
+}
+
+fn generate_colours(event_sink: druid::ExtEventSink) {
```

Comment:
Although I personally use international spellings frequently, we have decided to use `Color` instead of `Colour` in druid so I think it probably makes sense to be consistent with that here as well?

Reply:
oh sry, I try to keep is US English as well, but my nature is very much still with colour.

Reply:
me too, but sometimes we must defer to the imperialists. 😢